DiagnosticsExtension/Models/ConnectionStringValidator/BlobStorageValidator.cs

```diff
@@ -0,0 +1,139 @@
+// </copyright>
+// -----------------------------------------------------------------------
+
+using DiagnosticsExtension.Controllers;
+using DiagnosticsExtension.Models.ConnectionStringValidator.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Queues;
+using Azure.Storage.Blobs.Models;
+using Microsoft.WindowsAzure.Storage;
+
+namespace DiagnosticsExtension.Models.ConnectionStringValidator
+{
+    public class BlobStorageValidator : IConnectionStringValidator
+    {
+        public string ProviderName => "Microsoft.WindowsAzure.Storage";
+
+        public ConnectionStringType Type => ConnectionStringType.BlobStorageAccount;
+
+        ConnectionStringValidationResult.ManagedIdentityType identityType;
+
+        private const string ManagedIdentity = "managedIdentityCredentialMissing";
+
+        public async Task<ConnectionStringValidationResult> ValidateViaAppsettingAsync(string appsettingname, string entityName)
+        {
+            var response = new ConnectionStringValidationResult(Type);
+
+            try
+            {
+                var result = await TestConnectionStringViaAppSetting(appsettingname, entityName);
+                if (result.Succeeded)
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.Success;
+                }
+                else
+                {
+                    throw new Exception("Unexpected state reached: result.Succeeded == false is unexpected!");
+                }
+            }
+            catch (Exception e)
+            {
+                response = ConnectionStringResponseUtility.EvaluateResponseStatus(e, Type, identityType);
+            }
+
+            return response;
+        }
+
+        public async Task<TestConnectionData> TestConnectionStringViaAppSetting(string appSettingName, string entityName)
+        {
+            var envDict = Environment.GetEnvironmentVariables();
+            string appSettingClientIdValue, appSettingClientCredValue = null;            
+            BlobServiceClient client = null;
+
+            if (envDict.Contains(appSettingName))
+            {
+                try
+                {
+                    string connectionString = Environment.GetEnvironmentVariable(appSettingName);
+                    client = new BlobServiceClient(connectionString);
+                }
+                catch (ArgumentNullException e)
+                {
+                    throw new EmptyConnectionStringException(e.Message, e);
+                }
+                catch (Exception e)
+                {
+                    throw new MalformedConnectionStringException(e.Message, e);
+                }
+            }
+            else
+            {
+                try
+                {
+                    string serviceUriString = Environment.GetEnvironmentVariable(appSettingName + "__blobServiceUri");
```

Comment:
Just realized that we are missing the variant for AzureWebJobsStorage which is AzureWebJobsStorage_accountName and we may need to have special logic for this.  See here for details:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-identity-based-connections-tutorial#use-managed-identity-for-azurewebjobsstorage-preview